From 9ba1c0d713a813894fe2817db44d95a1af53c0a4 Mon Sep 17 00:00:00 2001
From: caoyi1 <caoyi1@lenovo.com>
Date: Tue, 24 Nov 2015 15:48:35 +0800
Subject: [PATCH 520/959] IKQCOMSD1-3075:input misc epl8802,modify dts and
 config

Change-Id: Iac0a56e3dbeb2d064879df05ee8c19b9eb33afd7
Signed-off-by: caoyi1 <caoyi1@motorola.com>
Reviewed-on: http://gerrit.mot.com/802381
SLTApproved: Slta Waiver <sltawvr@motorola.com>
SME-Granted: SME Approvals Granted
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Robert Wiegand <Robert.P.Wiegand@motorola.com>
Reviewed-by: Daniel Moore <dtmoore@motorola.com>
Reviewed-by: Igor Kovalenko <igork@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi | 35 +++++++++++++++++++++++++
 arch/arm/configs/ext_config/moto-msm8916.config |  1 +
 2 files changed, 36 insertions(+)

diff --git a/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi b/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
index 831ea42dc9a..ffd0c3d67a5 100644
--- a/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8952-moto-common.dtsi
@@ -309,6 +309,32 @@
 			bias-disable; /* No PULL */
 		};
 	};
+
+	epl8802_active {
+		/* interrupt */
+		qcom,pins = <&gp 43>;
+		qcom,num-grp-pins = <1>;
+		qcom,pin-func = <0>;
+		label = "epl8802-active";
+		/* active state */
+		epl8802_active: epl8802_active {
+			drive-strength = <2>; /* 2 MA */
+			bias-disable; /* No PULL */
+		};
+	};
+	epl8802_suspend {
+		/* interrupt  */
+		qcom,pins = <&gp 43> ;
+		qcom,num-grp-pins = <1>;
+		qcom,pin-func = <1>;
+		label = "epl8802-suspend";
+		/*suspended state */
+		epl8802_suspend: epl8802_suspend {
+			drive-strength = <2>; /* 2 MA */
+			bias-disable; /* No PULL */
+		};
+	};
+
 	usb_id_pin {
 		qcom,pins = <&gp 62>;
 		qcom,num-grp-pins = <1>;
@@ -864,6 +890,15 @@
 		pinctrl-1 = <&laser_suspend >;
 		st,sensorthreshold = <40 45>;
 	};
+
+	elan@49 {
+		compatible = "epl,epl8802";
+		reg = <0x49>;
+		epl,irq-gpio = <&msm_gpio 43 0x00>;
+		pinctrl-names = "epl8802_default", "epl8802_suspend";
+		pinctrl-0 = <&epl8802_active>;
+		pinctrl-1 = <&epl8802_suspend>;
+	};
 };
 
 &other_ext_mem {
diff --git a/arch/arm/configs/ext_config/moto-msm8916.config b/arch/arm/configs/ext_config/moto-msm8916.config
index 0beb8d9e8b3..31d5b192d69 100644
--- a/arch/arm/configs/ext_config/moto-msm8916.config
+++ b/arch/arm/configs/ext_config/moto-msm8916.config
@@ -53,5 +53,6 @@ CONFIG_MSM_TZ_LOG_WDOG_DUMP=y
 CONFIG_LEDS_ATC=y
 CONFIG_INPUT_KEYCHORD=y
 CONFIG_STMVL53L0=y
+CONFIG_EPL8802=y
 CONFIG_SENSORS_FPC_1020=y
 CONFIG_MMC_FFU=y
-- 
2.11.0

